### PR TITLE
Fixes changeling tentacle secondary behaviour

### LINKED ISSUE #69013: Changeling tentacle holds up the targets instead of stealing their items
## Round ID:
Different server

## Testmerges:

N/A

## Reproduction:

Activate your changeling tentacle, right click on someone. Instead of shooting out and stealing their active item, as described, it will instead initiate a hold up, slowly aiming at the target.

### PR DESCRIPTION
**Pull request by TheBoondock**
## About The Pull Request
Fixes #69013 
## Changelog
:cl:
fix: fixed using the changeling tentacle arm via right mouse click holding people up instead of firing it. Now it will fire like normal
/:cl:

### CODE REVIEW

File: code/modules/antagonists/changeling/powers/mutations.dm [hunk 1]
```diff
@@ -274,6 +274,10 @@
 /obj/item/gun/magic/tentacle/shoot_with_empty_chamber(mob/living/user as mob|obj)
 	to_chat(user, span_warning("The [name] is not ready yet."))
 
+/obj/item/gun/magic/tentacle/attack_secondary(mob/living/victim, mob/living/user, params)
+	. = ..()
```
Comment (Time-Green) [inline code/modules/antagonists/changeling/powers/mutations.dm:278]:
```suggestion
	..()
```
. =  sets the return value, which is pointless if you immediately return something else

**Review by Time-Green — CHANGES_REQUESTED** [review]

File: code/modules/antagonists/changeling/powers/mutations.dm [hunk 2]
```diff
@@ -274,6 +274,10 @@
 /obj/item/gun/magic/tentacle/shoot_with_empty_chamber(mob/living/user as mob|obj)
 	to_chat(user, span_warning("The [name] is not ready yet."))
 
+/obj/item/gun/magic/tentacle/attack_secondary(mob/living/victim, mob/living/user, params)
+	..()
+	return fire_gun(victim, user)
```
Comment (Time-Green) [inline code/modules/antagonists/changeling/powers/mutations.dm:279]: Did this work by the way? I think attack_secondary has specific return values it expects, and you also call the parent proc so wont this just hold someone up and then immediately shoot them?

Correct me if I’m wrong, I can’t check right now

Comment (TheBoondock) [reply code/modules/antagonists/changeling/powers/mutations.dm:279]: From testing with combat mode on it just shoot the tentacle instead. I can retest it once I'm home

Comment (Time-Green) [reply code/modules/antagonists/changeling/powers/mutations.dm:279]: I think it might act weird then in non-combat mode

**Comment by TheBoondock** [comment]
Right so I implemented another early return on `afterattack_secondary` and no_holdup for when we dont want that happening.

File: code/modules/projectiles/gun.dm [hunk 3]
```diff
@@ -189,9 +189,11 @@
 		for(var/obj/O in contents)
 			O.emp_act(severity)
 
-/obj/item/gun/afterattack_secondary(mob/living/victim, mob/living/user, params)
+/obj/item/gun/afterattack_secondary(mob/living/victim, mob/living/user, params, no_holdup = FALSE)
```
Comment (Time-Green) [inline code/modules/projectiles/gun.dm:192]: Sorry, I wasn’t clear. Please make this a variable on the gun itself, not a parameter. A parameter can be useful, but not in this case. It means that everyone that doesn’t want a hold_up has to override the proc, when a simple no_holdup = TRUE would’ve sufficed.

Also not really wrong but calling it can_holdup prevents people having to work with double negatives (like no_holdup = FALSE), which helps a bit with readability

**Pull request merged**